src/fentry.bpf.c

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+/* Copyright (c) 2021 Sartura */
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+SEC("fentry/bprm_execve")
+int BPF_PROG(bprm_execve, struct linux_binprm *bprm, int fd, struct filename *filename, int flags)
+{
+	pid_t pid;
+	const char *name = filename->name;
+
+	pid = bpf_get_current_pid_tgid();
+
+	bpf_printk("fentry: pid = %d, filename = %s\n", pid, name);
+	return 0;
+}
+
+SEC("fexit/bprm_execve")
+int BPF_PROG(bprm_execve_exit, struct linux_binprm *bprm, int fd, struct filename *filename, int flags, int ret)
+{
+	bpf_printk("fexit: ret = %d\n", ret);
```

Comment:
it would be great to demonstrate that you can actually read input arguments, so please output filename here as well. That's another huge and important difference between kretprobe and fexit (I might have forgotten to emphasize that, please add to the README as well). This is a critical feature in some cases.